services/retentionmgr/retention.go

```diff
@@ -696,21 +703,21 @@ func (t *RetentionManager) computeRetention(job *retentionJob, log bark.Logger)
 
 	// -- step 6: check to see if the extent status can be updated to 'consumed' -- //
 
-	// move the extent to 'consumed' if either:
+	// move the extent to 'consumed' if the extent is "sealed" _and_ either:
 	// A. all of the following are true:
-	// 	1. the extent was sealed
-	// 	2. the extent as fully consumed by all of the consumer groups
-	// 	3. a period of 'soft retention period' has passed (in other words,
+	// 	1. the extent was fully consumed by all of the consumer groups
+	// 	2. a period of 'soft retention period' has passed (in other words,
 	// 	   a consumer that is consuming along the soft retention time has
 	//	   "consumed" the extent)
 	// B. or, the hard-retention has reached the end of the sealed extent,
 	// 	in which case we will force the extent to be "consumed"
-	// NB: retentionAddr == ADDR_BEGIN indicates there was an error, so we no-op
-	if job.retentionAddr != store.ADDR_BEGIN &&
-		((ext.status == shared.ExtentStatus_SEALED &&
-			minAckAddr == store.ADDR_SEAL &&
-			softRetentionConsumed) ||
-			hardRetentionConsumed) {
+	// NB: if there was an error querying either the hard/soft retention addresses,
+	// {soft,hard}RetentionConsumed would be set to 'false'; if there was an error
+	// querying ack-addr, then allHaveConsumed will be false. therefore errors in
+	// either of the conditions would cause the extent to *not* be moved to the
+	// CONSUMED state, and would cause it to be retried on the next iteration.
+	if (ext.status == shared.ExtentStatus_SEALED) &&
+		((allHaveConsumed && softRetentionConsumed) || hardRetentionConsumed) {
```

Comment:
I agree .. I will special-case when there are no consumers and if this is a remote extent, to not move the extent to consumed.

Note: this diff does not affect the computation of the retention-address (it only affects when we move this to consumed). I shall address that as part of a separate change, for the zero-consumer case.